gpuflow/dataplane/gpu/cuda/cuda_async_lcore_function.cu

```diff
@@ -64,6 +68,48 @@ __device__ void IPv6Processing(CustomEtherIPHeader *custom_ether_ip_header, IPv6
   // 6. Assign the next hop to the custom_ether_ip_header->dst_port.
   // 7. If no rules matched,
   // 8. Assign the custom_ether_ip_header->dst_port = 254
+  
+  unsigned long int ipv6_addr_first24 = ipv6_addr_le[0] << 16 | ipv6_addr_le[1] << 8 | ipv6_addr_le[2];
+  if ((lpm6_table_ptr + ipv6_addr_first24) != nullptr) {
+    IPv6RuleEntry *entry = lpm6_table_ptr + ipv6_addr_first24;
+    if (entry->valid_flag) {
+      // Match first 24 bits
+      if (entry->external_flag) {
+        // TODO: Lookup tbl8 table.
+      } else {
+        // Do not have to lookup tbl8 table
+        EtherCopy(&custom_ether_ip_header->ether_header, port_id, entry->next_hop, dev_mac_addresses_array);
+        custom_ether_ip_header->dst_port = entry->next_hop;
+      }
+    } else {
+      // Do not exactly match first 24 bits
+      uint8_t shift = 0;
+      bool matched = false;
+
+      // Go through first 23, 22, 21, ......, 0 bits.
+      for(uint8_t i = 0; i < 24; i++) {
+        shift = i + 1;
```

Comment:
You don't need to check like this. In the setup rules phase, we already fill all related rules in the lpm table.
i.e. in a 4-bit context, `b1000` with 2 bits netmask, we already fill the `b1000`, `b1001`, `b1010`, `b1011` in the lpm table. Therefore, any addresses in the context can be directly indexed.

That is, we only need to check 24 bits.

Reply:
```c++
unsigned long int ipv6_addr_first24 = ipv6_addr_le[0] << 16 | ipv6_addr_le[1] << 8 | ipv6_addr_le[2];
if ((lpm6_table_ptr + ipv6_addr_first24) != nullptr) {
  IPv6RuleEntry *entry = lpm6_table_ptr + ipv6_addr_first24;
  if (entry->valid_flag) {
    // Match first 24 bits
    if (entry->external_flag) {
      // TODO: Lookup tbl8 table.
    } else {
      // Do not have to lookup tbl8 table
      EtherCopy(&custom_ether_ip_header->ether_header, port_id, entry->next_hop, dev_mac_addresses_array);
      custom_ether_ip_header->dst_port = entry->next_hop;
   } else {
  // Not matched, ignore.
  }
}

```